Playwright CT mount fixture type triggers @typescript-eslint/unbound-method when destructured

## Summary

Destructuring `mount` from the Playwright component-test fixture callback triggers a false positive from `@typescript-eslint/unbound-method`.

This appears to come from the published `@playwright/experimental-ct-vue` type declarations rather than from runtime behavior. The documented Playwright CT pattern looks like this:

```ts
import { test } from '@playwright/experimental-ct-vue';

test('reproduces the mount typing false positive', async ({ mount }) => {
  await mount({});
});
```

## Reproduction

Repro repo: https://github.com/rotu/playwright-ct-unbound-method-repro

Steps:

1. `git clone https://github.com/rotu/playwright-ct-unbound-method-repro.git`
2. `cd playwright-ct-unbound-method-repro`
3. `npm install`
4. `npm run lint`

## Actual

ESLint reports:

```text
src/repro.spec.ts
  5:61  error  A method that is not declared with `this: void` may cause unintentional scoping of `this` when separated from its object.
Consider using an arrow function or explicitly `.bind()`ing the method to avoid calling the method with an unintended `this` value.
If a function does not access `this`, it can be annotated with `this: void`  @typescript-eslint/unbound-method
```

## Expected

The documented `async ({ mount }) => { ... }` CT fixture pattern should not trigger `@typescript-eslint/unbound-method`.

## Versions

- `@playwright/experimental-ct-vue`: `1.59.1`
- `@typescript-eslint/eslint-plugin`: `8.31.1`
- `typescript`: `5.8.3`

--- a/packages/playwright-ct-core/src/mount.ts
+++ b/packages/playwright-ct-core/src/mount.ts
@@ -27,8 +27,8 @@ import type { Page as PageImpl } from 'playwright-core/lib/client/page';
 let boundCallbacksForMount: Function[] = [];
 
 interface MountResult extends Locator {
-  unmount(locator: Locator): Promise<void>;
-  update(options: Omit<MountOptions, 'hooksConfig'> | string | JsxComponent): Promise<void>;
+  unmount: () => Promise<void>;
+  update: (options: ObjectComponentOptions | JsxComponent) => Promise<void>;
 }
 
 type TestFixtures = PlaywrightTestArgs & PlaywrightTestOptions & {

--- a/packages/playwright-ct-react/index.d.ts
+++ b/packages/playwright-ct-react/index.d.ts
@@ -23,15 +23,15 @@ export interface MountOptions<HooksConfig> {
 }
 
 export interface MountResult extends Locator {
-  unmount(): Promise<void>;
-  update(component: React.JSX.Element): Promise<void>;
+  unmount: () => Promise<void>;
+  update: (component: React.JSX.Element) => Promise<void>;
 }
 
 export interface ComponentFixtures {
-  mount<HooksConfig>(
+  mount: <HooksConfig>(
     component: React.JSX.Element,
     options?: MountOptions<HooksConfig>
-  ): Promise<MountResult>;
+  ) => Promise<MountResult>;
 }
 
 export const test: TestType<ComponentFixtures>;

--- a/packages/playwright-ct-react17/index.d.ts
+++ b/packages/playwright-ct-react17/index.d.ts
@@ -21,15 +21,15 @@ export interface MountOptions<HooksConfig> {
 }
 
 export interface MountResult extends Locator {
-  unmount(): Promise<void>;
-  update(component: JSX.Element): Promise<void>;
+  unmount: () => Promise<void>;
+  update: (component: JSX.Element) => Promise<void>;
 }
 
 export interface ComponentFixtures {
-  mount<HooksConfig>(
+  mount: <HooksConfig>(
     component: JSX.Element,
     options?: MountOptions<HooksConfig>
-  ): Promise<MountResult>;
+  ) => Promise<MountResult>;
 }
 
 export const test: TestType<ComponentFixtures>;

--- a/packages/playwright-ct-vue/index.d.ts
+++ b/packages/playwright-ct-vue/index.d.ts
@@ -39,28 +39,30 @@ export interface MountOptionsJsx<HooksConfig> {
 }
 
 export interface MountResult<Component> extends Locator {
-  unmount(): Promise<void>;
-  update(options: {
+  unmount: () => Promise<void>;
+  update: (options: {
     props?: Partial<ComponentProps<Component>>;
     slots?: Partial<ComponentSlots>;
     on?: Partial<ComponentEvents>;
-  }): Promise<void>;
+  }) => Promise<void>;
 }
 
 export interface MountResultJsx extends Locator {
-  unmount(): Promise<void>;
-  update(component: JSX.Element): Promise<void>;
+  unmount: () => Promise<void>;
+  update: (component: JSX.Element) => Promise<void>;
 }
 
 export interface ComponentFixtures {
-  mount<HooksConfig>(
-    component: JSX.Element,
-    options?: MountOptionsJsx<HooksConfig>
-  ): Promise<MountResultJsx>;
-  mount<HooksConfig, Component = unknown>(
-    component: Component,
-    options?: MountOptions<HooksConfig, Component>
-  ): Promise<MountResult<Component>>;
+  mount: {
+    <HooksConfig>(
+      component: JSX.Element,
+      options?: MountOptionsJsx<HooksConfig>
+    ): Promise<MountResultJsx>;
+    <HooksConfig, Component = unknown>(
+      component: Component,
+      options?: MountOptions<HooksConfig, Component>
+    ): Promise<MountResult<Component>>;
+  };
 }
 
 export const test: TestType<ComponentFixtures>;

